src/ai/backend/agent/server.py

```diff
@@ -827,15 +827,32 @@ async def server_main(
 @click.option(
     "--debug",
     is_flag=True,
-    help="Enable the debug mode and override the global log level to DEBUG.",
+    help="This option will soon change to --log-level TEXT option.",
+)
+@click.option(
+    "--log-level",
+    default="info",
+    help="Choose logging level from... debug, info, warning, error, critical",
 )
 @click.pass_context
 def main(
     cli_ctx: click.Context,
     config_path: Path,
-    debug: bool,
+    log_level: str,
+    debug: bool = False,
 ) -> int:
 
+    if debug:
+        click.echo("Please use --log-level options instead")
+        click.echo("--debug options will soon change to --log-level TEXT option.")
+        log_level = "debug"
+
+    if log_level not in ["debug", "info", "warning", "error", "critical"]:
+        click.echo("Undefined log-level")
+        click.echo("Try 'backend.ai ag start-server -h' for help")
+        exit(1)
```

Comment:
Bit of a combination of @fregataa and @rapsealk 's review. How about muxing log levels as an enum and using click's `Choice` option? That way we can relieve the burden to manually validate log level string user provided.
Ref: https://github.com/pallets/click/issues/605#issuecomment-231559324